File: solutions/lua/grains/1/grains.lua
local grains = {}

function grains.square(n)
  return (
    n == 1 and 1 or 2 ^ (n - 1)
  )
end

function grains.total()
  local sum = 0
  local grains = 1.0
  for i = 1, 64 do
    sum = sum + grains
    grains = grains * 2
  end
  return sum
end

return grains

File: solutions/go/grains/1/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	if number <= 0 || number > 64 {
		return 0, errors.New("")
	}
	return uint64(math.Pow(2, float64(number-1))), nil
}

func Total() (result uint64) {
	for i := 1; i <= 64; i++ {
		n, _ := Square(i)
		result += n
	}
	return result
}
